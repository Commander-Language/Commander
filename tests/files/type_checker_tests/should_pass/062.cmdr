//Test that negate operations type check
x = 2;
y = -[1, 2, 3][x];
