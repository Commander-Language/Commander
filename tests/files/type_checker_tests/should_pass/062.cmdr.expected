(PrgmNode
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (IntExprNode 2 INT) INT))
	(ExprStmtNode (BinOpExprNode (BindingNode y) SET (UnOpExprNode (IndexExprNode (ArrayExprNode (ExprsNode (IntExprNode 1 INT) (IntExprNode 2 INT) (IntExprNode 3 INT)) INT[]) (LValueExprNode (VarExprNode x INT) INT) INT) NEGATE INT) INT)))
